netsync/manager.go

```diff
@@ -1008,7 +1009,17 @@ func (sm *SyncManager) handleHeadersMsg(hmsg *headersMsg) {
 			peer.Disconnect()
 			return
 		}
-
+		// Processing the block headers that are downloaded and if it passes
+		// all the checks, creating a block node that only contains header.
+		err := chain.ProcessBlockHeader(blockHeader)
+		if err != nil {
+			// Note that there is no need to check for an orphan header here
+			// because they were already verified to connect above.
+			log.Debugf("Failed to process block header %s from peer %s: %v -- "+
+				"disconnecting", blockHeader.BlockHash(), peer, err)
+			peer.Disconnect()
+			return
+		}
 		// Verify the header at the next checkpoint height matches.
 		if node.height == sm.nextCheckpoint.Height {
```

Comment:
We wouldn't need this check here because it's redundant with the checkpoint check in `b.checkHeaderContext` that's called in `maybeAcceptBlockHeader`, which is called from `ProcessBlockHeader`.

https://github.com/utreexo/utreexod/blob/53dc2e1096418f7286cb29e7d0b679c3129a524a/blockchain/validate.go#L676-L697